server/src/main/java/org/opensearch/index/remote/RemoteStoreUtils.java

```diff
@@ -146,4 +148,25 @@ static String longToCompositeBase64AndBinaryEncoding(long value, int len) {
         assert base64DecimalValue >= 0 && base64DecimalValue < 64;
         return URL_BASE64_CHARSET[base64DecimalValue] + binaryPart;
     }
+
+    /**
+     * Determines the remote store path strategy by reading the custom data map in IndexMetadata class.
+     */
+    public static RemoteStorePathStrategy determineRemoteStorePathStrategy(IndexMetadata indexMetadata, boolean assertRemoteCustomData) {
+        Map<String, String> remoteCustomData = indexMetadata.getCustomData(IndexMetadata.REMOTE_STORE_CUSTOM_KEY);
+        if (assertRemoteCustomData) {
+            assert remoteCustomData == null || remoteCustomData.containsKey(RemoteStoreEnums.PathType.NAME);
+        }
```

Comment:
With the assert we check that the remote custom data is either null or remote custom data contains the key `path_type`. This currently holds true for index creation & path upload too, but with introduction of other keys in remote custom data (for instance translog ckp optimisation), it is possible that this might not hold true. Hence, add tests such as empty remote custom data / remote custom data with no key that will hold true even in future and thus this separation.